x-pack/libbeat/common/aws/credentials.go

```diff
@@ -44,10 +45,21 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
 	// If accessKeyID, secretAccessKey or sessionToken is not given, then load from default config
 	// Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
 	// with more details.
+	// If credential_profile_name is empty, then default profile is used.
+	var options []external.Config
 	if config.ProfileName != "" {
-		return external.LoadDefaultAWSConfig(
-			external.WithSharedConfigProfile(config.ProfileName),
-		)
+		options = append(options, external.WithSharedConfigProfile(config.ProfileName))
+	}
+	// If shared_credential_file is empty, then external.LoadDefaultAWSConfig
+	// function will load AWS config from current user's home directory.
+	// Linux/OSX: "$HOME/.aws/credentials"
+	// Windows:   "%USERPROFILE%\.aws\credentials"
+	if config.SharedCredentialFile != "" {
+		options = append(options, external.WithSharedConfigFiles([]string{config.SharedCredentialFile}))
+	}
+
+	if options != nil {
+		return external.LoadDefaultAWSConfig(options...)
```

Comment:
I think it works also if `options` is nil, so this if wouldn't be needed.

Reply:
Yeah I tested yesterday and thought it didn't work with options=nil so I added this if statement. But I tested again just now and options=nil works for LoadDefaultAWSConfig. Will remove this if statement. Thanks!